src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -638,7 +638,7 @@ def handleCondorStatusService(self):
         if isCMSSWSupported(self.getCmsswVersion(), "CMSSW_7_6_0"):
             self.logger.info("Tag chirp updates from CMSSW with step %s", self.step.data._internal_name)
             self.process.add_(cms.Service("CondorStatusService",
-                                          tag=cms.untracked.string("_%s_" % self.step.data._internal_name)))
+                                          tag=cms.untracked.string("_%s_" % encodeUnicodeToBytes(self.step.data._internal_name))))
```

Comment:
@amaltaro are you 100% sure of this change? because in py3 it would actually add `b''` to the string

```plaintext
> python3
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> d = "hello world"
>>> "_%s_" % d.encode()
"_b'hello world'_"
```

Reply:
If this was intended to remove the `u''` in py2, then we may use `encodeUnicodeToBytesConditional(..., condition=PY2)`

Reply:
This patch is not being provided against the master branch. It's only a patch for the current - quite old - agents that we have in production. We do not even have the `encodeUnicodeToBytesConditional` function on this branch :-D

so, I believe it's safe enough. I did not manage to go that far in the Python3 WMAgent, so I do not know right now whether something "similar" would be required against master.

Reply:
Uh! you are correct! If this is actually only intended to be run in py2, It's safe! thanks Alan! Sorry for bugging :)